include/ppx/knob.h

```diff
@@ -0,0 +1,284 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ppx_knob_h
+#define ppx_knob_h
+
+#include "ppx/command_line_parser.h"
+#include "ppx/config.h"
+#include "ppx/log.h"
+
+#include <functional>
+#include <iostream>
+#include <unordered_set>
+#include <vector>
+
+namespace ppx {
+
+// ---------------------------------------------------------------------------------------------
+// Knob Classes
+// ---------------------------------------------------------------------------------------------
+
+// Knob is an abstract class which contains common features for all knobs and knob hierarchy
+class Knob
+{
+public:
+    Knob(const std::string& displayName, const std::string& flagName, const std::string& flagDesc)
+        : mDisplayName(displayName), mFlagName(flagName), mFlagDesc(flagDesc) {}
+
+    std::string GetDisplayName() const { return mDisplayName; }
+    std::string GetFlagName() const { return mFlagName; }
+    std::string GetFlagDesc() const { return mFlagDesc; }
+
+    // Child knobs are displayed in the UI below their parent and slightly indented
+    void               AddChild(Knob* pChild) { mChildren.push_back(pChild); }
+    std::vector<Knob*> GetChildren() const { return mChildren; }
+
+    virtual void Draw() = 0;
+
+    // Flag description in usage message
+    virtual std::string FlagText() const = 0;
```

Comment:
This function name should be more explicit. Something like GetKnobHelpText() or GetKnobFlagHelpText()